bin/improver-regrid

```diff
@@ -0,0 +1,113 @@
+                               {'default': False,
+                                'action': 'store_true',
+                                'help': ('If True, regridding will be '
+                                         'performed using iris.analysis.'
+                                         'Nearest() instead of Linear(). '
+                                         'Use for less continuous fields, e.g.'
+                                         ' precipitation.')}),
+                              (['--extrapolation_mode'],
+                               {'default': 'nanmask',
+                                'help': ('Mode to use for extrapolating data '
+                                         'in to regions beyond the limits of '
+                                         'the source_data domain. Modes are: '
+                                         'extrapolate - The extrapolation '
+                                         'points will take their value from '
+                                         'the nearest source point. '
+                                         'nan - The extrapolation points will '
+                                         'be be set to NaN. error - A '
+                                         'ValueError exception will be raised,'
+                                         ' notifying an attempt to extrapolate'
+                                         '. mask  - The extrapolation points '
+                                         'will always be masked, even if the '
+                                         'source data is not a MaskedArray. '
+                                         'nanmask - If the source data is a '
+                                         'MaskedArray the extrapolation points'
+                                         ' will be masked. Otherwise they will'
+                                         ' be set to NaN. Defaults to nanmask.'
+                                         )}),
+                              ]
+
+    cli_definition = {'central_arguments': [],
+                      'specific_arguments': cli_specific_arguments,
+                      'description': ('Regrid data from source_data on to the '
+                                      'grid contained within target_grid using'
+                                      ' iris.analysis.Linear() or optionally '
+                                      'iris.analysis.Nearest()')}
+
+    args = ArgParser(**cli_definition).parse_args()
+
+    source_data = load_cube(args.source_data_filepath)
+    target_grid = load_cube(args.target_grid_filepath)
+
+    regridder = iris.analysis.Linear(
+        extrapolation_mode=args.extrapolation_mode)
+    if args.nearest:
+        regridder = iris.analysis.Nearest(
+            extrapolation_mode=args.extrapolation_mode)
+
+    source_on_target = source_data.regrid(target_grid, regridder)
+    delete_attributes(source_on_target, ['mosg__grid', 'title'])
```

Comment:
Deleting them including domain? Maybe you could replace `mosg__grid_domain` with the target grid value if it exists.

Reply:
I did that originally, but after some discussion it wasn't clear that we will still have this metadata by the time we get to the other end of IMPROVER (level-3) anyway. Any thoughts on that? The cube still contains the coordinate system etc, just not the stage grid identifier.

Reply:
I think we should assume that Level 3 shares that piece of metadata with the Level 1, if only to make other systems lives easier. It's probably nicer for them to catalogue based on an attribute than delving into the grid mapping.

Ultimately we probably want a site-configurable list of special attributes, given that `mosg` is our own namespace rather than a generic one...